src/main.ts

```diff
@@ -82,6 +94,21 @@ const getUpdateStatus = async (uuid: string): Promise<any> => {
 	}
 };
 
+const checkLock = async (): Promise<boolean> => {
```

Comment:
This is only checking for locks in this service container, which will never exist. We need to check for locks in any service somehow. Specifically the user application service that is already creating locks. The supervisor does this, there are probably a few ways to solve this from inside a service.

Reply:
The supervisor uses a separate lockfile binary. We could use the same.